src/server/api.js

```diff
@@ -63,16 +63,42 @@ function submitApplication(req, res) {
       if (err) {
         debug(`voaService response had error ${err}`);
         // TODO(awong): This may leak server config info on error. Is that a problem?
-        res.status(500).send({ error: err });
+        res.status(500).json({ error: err });
```

Comment:
Yup...I think Robbie was saying node in send() attempts to sniff the
content type.

What I'm wondering is if it's better to disable the sniffing by explicitly
using json().

On Tue, May 24, 2016 at 2:15 PM, Joseph Stewart <notifications@github.com>
wrote:

> In src/server/api.js
> <https://github.com/department-of-veterans-affairs/healthcare-application/pull/86#discussion_r64470943>
> :
>
> > @@ -63,16 +63,42 @@ function submitApplication(req, res) {
> >        if (err) {
> >          debug(`voaService response had error ${err}`);
> >          // TODO(awong): This may leak server config info on error. Is that a problem?
> > -        res.status(500).send({ error: err });
> > +        res.status(500).json({ error: err });
>
> I tinkered with changing between .json() and .send() using curl as a
> client and the Chrome Boomerang app, and there was no observable
> difference. Both returned identical strings from JSON objects and both had *Content-Type:
> application/json* in the returned headers.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub
> <https://github.com/department-of-veterans-affairs/healthcare-application/pull/86/files/e19f2e126198f61ee1b63265740e89b451346737#r64470943>
>